plugins/parsers/value/parser.go

```diff
@@ -17,12 +16,12 @@ type ValueParser struct {
 }
 
 func (v *ValueParser) Parse(buf []byte) ([]telegraf.Metric, error) {
+	buf = bytes.TrimSpace(bytes.Trim(buf, "\x00"))
+
 	// unless it's a string, separate out any fields in the buffer,
 	// ignore anything but the last.
-	var vStr string
-	if v.DataType == "string" {
-		vStr = strings.TrimSpace(string(buf))
-	} else {
+	vStr := string(buf)
```

Comment:
rather than reassigning the variable, you can just do `vStr := string(bytes.TrimSpace(bytes.Trim(buf, "\x00")))`